src/pages/instances/actions/ConvertToInstanceBtn.tsx

```diff
@@ -0,0 +1,321 @@
+import usePortal from "react-useportal";
+import { UploadState } from "types/storage";
+import ProgressBar from "components/ProgressBar";
+import {
+  getSupportedArchitectures,
+  humanFileSize,
+  sendFileByWebSocket,
+} from "util/helpers";
+import StoragePoolSelector from "pages/storage/StoragePoolSelector";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { useEventQueue } from "context/eventQueue";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { getInstanceName } from "util/operations";
+
+export interface ConvertToInstanceFormValues {
+  imageFile: File | null;
+  name: string;
+  pool: string;
+  formatConversion: boolean;
+  virtioConversion: boolean;
+  architecture: string;
+}
+
+const CHUNK_SIZE = 1024 * 1024;
+
+const convertToInstancePayload = (values: ConvertToInstanceFormValues) => {
+  const fileSize = values.imageFile?.size || 0;
+  const conversionOptions = [];
+  if (values.formatConversion) {
+    conversionOptions.push("format");
+  }
+  if (values.virtioConversion) {
+    conversionOptions.push("virtio");
+  }
+
+  const instanceArgs = {
+    source: {
+      type: "conversion",
+      mode: "push",
+      conversion_options: conversionOptions,
+    },
+    devices: {
+      root: {
+        path: "/",
+        type: "disk",
+        pool: values.pool,
+      },
+    },
+    type: "virtual-machine",
+    name: values.name,
+    sourceDiskSize: fileSize,
+    architecture: values.architecture,
+  };
+
+  return JSON.stringify(instanceArgs);
+};
+
+const ConvertToInstanceBtn: FC = () => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const { project, isLoading: isProjectLoading } = useProject();
+  const instanceNameAbort = useState<AbortController | null>(null);
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+  const toastNotify = useToastNotification();
+  const navigate = useNavigate();
+  const eventQueue = useEventQueue();
+  const queryClient = useQueryClient();
+
+  const handleUploadProgress = (current: number, total: number) => {
+    setUploadState({
+      percentage: Math.floor((current / total) * 100),
+      loaded: current,
+      total: total,
+    });
+
+    if (current === total) {
+      handleCloseModal();
+      navigate(`/ui/project/${project?.name}/instances`);
+    }
+  };
+
+  const handleUploadError = (error: Error) => {
+    toastNotify.failure("Image upload failed.", error);
+  };
+
+  const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("imageFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      void formik.setFieldValue("imageFile", file);
+    }
+  };
+
+  const handleSuccess = (instanceName: string) => {
+    const message = (
+      <>
+        Created instance <strong>{instanceName}</strong>.
```

Comment:
I think we should link the instance name. Not sure if we need the configure link.

Reply:
The thinking here is to keep the behaviour consistent with uploading an instance in `UploadInstanceBtn`. The lxd-migrate tool also allows the user to configure the instance during the migration setup steps, which I think would align with this behaviour?

Reply:
I see, makes sense.

I think we should always link the instance name. I was conditioned to click on the name and got confused why in this occasion it is not clickable.